src/shogun/classifier/mkl/MKL.cpp

```diff
@@ -262,6 +264,49 @@ CMKL::~CMKL()
 	SG_UNREF(svm);
 }
 
+void CMKL::register_params()
+{
+	// Register parameters
+	SG_ADD((CSGObject**) &svm, "svm", "wrapper svm", MS_NOT_AVAILABLE);
+	SG_ADD(&C_mkl, "C_mkl", "C mkl", MS_NOT_AVAILABLE);
+	SG_ADD(&mkl_norm, "mkl_norm", "norm used in mkl", MS_NOT_AVAILABLE);
+	SG_ADD(&ent_lambda, "ent_lambda", "elastic net sparsity trade-off parameter", MS_NOT_AVAILABLE);
+	SG_ADD(&mkl_block_norm, "mkl_block_norm", "mkl sparse trade-off parameter", MS_NOT_AVAILABLE);
+	m_parameters->add_vector(&beta_local, &beta_local_size, "beta_local", "subkernel weights on L1 term of elastic net mkl");
+	SG_ADD(&mkl_iterations, "mkl_iterations", "number of mkl steps", MS_NOT_AVAILABLE);
+	SG_ADD(&mkl_epsilon, "mkl_epsilon", "mkl epsilon", MS_NOT_AVAILABLE);
+	SG_ADD(&interleaved_optimization, "interleaved_optimization", "whether to use mkl wrapper or interleaved opt.", MS_NOT_AVAILABLE);
+	SG_ADD(&w_gap, "w_gap", "gap between interactions", MS_NOT_AVAILABLE);
+	SG_ADD(&rho, "rho", "objective after mkl iterations", MS_NOT_AVAILABLE);
+	SG_ADD(&lp_initialized, "lp_initialized", "if lp is Initialized", MS_NOT_AVAILABLE);
+
```

Comment:
registered parameters need to get default parameters. Lots of problems incoming otherwise.

It is a different story for existing code where parameters were already registered, but as you are adding that here, you need to put default values for all of them in this method. Consequently, need to replace all those init fields in the constructor with explicit assignments